server/services/offender/responses/nextVisit.js

```diff
@@ -9,58 +8,30 @@ const {
   },
 } = require('../../../utils/enums');
 const { formatDateOrDefault } = require('../../../utils/date');
+const { fullNameOrDefault } = require('../../../utils/string');
 
-class NextVisit {
-  constructor(options = {}) {
-    this.startTime = options.startTime;
-    this.endTime = options.endTime;
-    this.status = options.status;
-    this.visitorName = options.visitorName;
-    this.visitType = options.visitType;
+module.exports = function NextVisit(response = []) {
+  if (!response?.length) {
+    return { hasNextVisit: false };
   }
 
-  hasNextVisit() {
-    return this.startTime != null;
-  }
-
-  format() {
-    return {
-      hasNextVisit: this.hasNextVisit(),
-      nextVisit: formatDateOrDefault(DEFAULT, LONG_PRETTY_DATE, this.startTime),
-      nextVisitDay: formatDateOrDefault(DEFAULT, PRETTY_DAY, this.startTime),
-      nextVisitDate: formatDateOrDefault(
-        DEFAULT,
-        PRETTY_DAY_AND_MONTH,
-        this.startTime,
-      ),
-      visitorName: this.visitorName
-        ? capitalizePersonName(this.visitorName)
-        : DEFAULT,
-      visitType: this.visitType ? this.visitType.split(' ').shift() : DEFAULT,
-      startTime: formatDateOrDefault(DEFAULT, PRETTY_TIME, this.startTime),
-      endTime: formatDateOrDefault(DEFAULT, PRETTY_TIME, this.endTime),
-    };
-  }
+  const { visitDetails, visitors } = response[0];
+  const { startTime, endTime, visitTypeDescription: visitType } = visitDetails;
 
-  static from(response = {}) {
-    const {
+  return {
+    hasNextVisit: startTime != null,
+    nextVisit: formatDateOrDefault(DEFAULT, LONG_PRETTY_DATE, startTime),
```

Comment:
if we know that this always represents a valid next visit then don't need to worry about defaulting